src/NeoSharp.Core/Network/Server.cs

```diff
@@ -45,12 +46,18 @@ public class Server : IServer, IDisposable
             IServerContext serverContext,
             IPeerMessageListener peerMessageListener)
         {
-            if (config == null) throw new ArgumentNullException(nameof(config));
-            _peerFactory = peerFactory ?? throw new ArgumentNullException(nameof(peerFactory));
-            _peerListener = peerListener ?? throw new ArgumentNullException(nameof(peerListener));
-            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
-            _serverContext = serverContext ?? throw new ArgumentNullException(nameof(serverContext));
-            _peerMessageListener = peerMessageListener ?? throw new ArgumentNullException(nameof(peerMessageListener));
+            Guard.IsNotNull(config, nameof(config));
```

Comment:
I used this technique before but these days C# is more convenient for input checks. Here are my arguments against:
1. Assignment and input verification are splitted. It doubles lines number the dev needs to write and creates confusion because to understand what was check and not assigned or vice versa the dev needs to check two places.
2. Everything is going to be dependent on Guard class and namespace it is in.
3. Extra call of the static method. I am not sure it will be inlined.
4. The check ?? throw new Exception is generated by Resharper what speeds up the development. In the case of Guard class, it is dev responsibility.

Reply:
I've hard this arguments ... 

check this link
http://blog.ploeh.dk/2014/08/07/why-dry/

Reply:
Oh, jeez... Anybody else but Mark Seemann. I have not fully recovered mentally after customizing AutoFixture 😆 Please, note it was published in 2014. At that time even I used Guard pattern because it was slick. But the C# did not stop the development and today you do not need to write __if__ statements or to pass a param name as a string. Just ?? and nameof operators.